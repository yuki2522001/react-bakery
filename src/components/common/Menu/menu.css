.menu__list {
  display: flex;
}

.menu__item {
  list-style: none;
  cursor: pointer;
  font-size: 14px;
  margin-right: 30px;
}

.menu__item > a {
  text-decoration: none;
  text-transform: uppercase;
  font-weight: bold;
  color: var(--text-master);
}