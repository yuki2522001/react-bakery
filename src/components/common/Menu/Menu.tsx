import React, { memo } from "react";
import { Link } from "react-router-dom";
import { MenuProps, MenuType } from "@common-types/menu";
import "./menu.css";

const Menu: React.FC<MenuProps> = memo(({ menuList }) => {
  function renderMenuList(list: MenuType[]) {
    return list.map((item) => (
      <li className="menu__item" key={item.key}>
        <a href="">{item.label}</a>
      </li>
    ));
  }

  return (
    <div data-testid="menu" className="menu">
      <ul className="menu__list">
        <li className="menu__item">
          <Link to="/">Home</Link>
        </li>
        {renderMenuList(menuList)}
      </ul>
    </div>
  );
});

export default Menu;
