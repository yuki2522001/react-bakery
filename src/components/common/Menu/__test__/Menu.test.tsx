import { render } from "@testing-library/react";
import MENU_LIST from "../../../../constants/menu";
import "@testing-library/jest-dom";
import Menu from "../Menu";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

describe("Menu component", () => {
  test("should render menu component", () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <Menu menuList={MENU_LIST} />
      </Router>
    );
    expect(getByTestId("menu")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router location={history.location} navigator={history}>
        <Menu menuList={MENU_LIST} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
