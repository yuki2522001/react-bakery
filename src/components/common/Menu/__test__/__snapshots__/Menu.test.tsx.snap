// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="menu"
    data-testid="menu"
  >
    <ul
      class="menu__list"
    >
      <li
        class="menu__item"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="menu__item"
      >
        <a
          href=""
        >
          About
        </a>
      </li>
      <li
        class="menu__item"
      >
        <a
          href=""
        >
          Shop
        </a>
      </li>
      <li
        class="menu__item"
      >
        <a
          href=""
        >
          Blog
        </a>
      </li>
      <li
        class="menu__item"
      >
        <a
          href=""
        >
          FAQ
        </a>
      </li>
      <li
        class="menu__item"
      >
        <a
          href=""
        >
          Contact
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
